connectors/sources/network_drive.py

```diff
@@ -485,6 +485,10 @@ async def fetch_file_content(self, path):
         """
         self._logger.debug(f"Fetching the contents of file on path: {path}")
         try:
+            loop = asyncio.get_running_loop()
+            await loop.run_in_executor(
+                executor=None, func=self.smb_connection.create_connection
+            )
```

Comment:
Let's be consistent and `await asyncio.to_thread` everywhere

Reply:
Additionally - the connection should have already be created here, right?

Reply:
No, the pull from main, added merge changes from PR of @akanshi-elastic related to local content extraction feature for Network Drive, which introduced a change in the way files would be downloaded in the connector.

I took a call with @akanshi-elastic and we found that:
Previously, the connector would itself call fetch_file_content : https://github.com/elastic/connectors/blob/8ce8283abee1ca1ce70eec4bca026d77cda66771/connectors/sources/network_drive.py#L494, so the connection would always be existing.
However, after the changes of local extraction PR, the method `fetch_file_content` is now passed as download_func to `download_and_extract_file` and then it's called outside the connector: https://github.com/elastic/connectors/commit/9b9609da683ca429d8e19fd1beeff35182b6d8a6#diff-29e12700a1a4cf24af22d5bf0a870c9ebfede0ec41dc991406b951d0100393ffR501, so it is not guaranteed that the connection would be existing.

Reply:
But this is something I do not get - download cannot happen without us listing the objects, so how can we download something without meeting it when listing files/directories?

Reply:
updated it and removed the create connection from here 👍